modules/axhal/src/arch/x86_64/mod.rs

```diff
@@ -2,7 +2,7 @@ mod context;
 mod gdt;
 mod idt;
 
-#[cfg(target_os = "none")]
+// #[cfg(target_os = "none")]
 mod trap;
```

Comment:
Add the `cfg` back.

Reply:
OK.

Reply:
Sorry, we need to remove the `cfg` line to pass CI

Reply:
If i add back the `cfg`, i cannot access `IRQ_VECTOR_END` and `IRQ_VECTOR_START` in `mod trap`, thus failed to pass ci.
Since I don't know when `target_os = "none"` is set, can you give me some suggestions?😚

Reply:
If we build with the target `x86_64-unknown-none`, `target_os = "none"` is set. Otherwise, if we run
`cargo build`, `cargo clippy`, and `cargo test` without the `--target` argument, `target_os` is your host OS (usually `linux`).

In this case, we added the `cfg` to prevent trap handler when build as a host program (`target_os = "linux"`), but it is actually no need. So just remove the `cfg` line.